src/sentry/plugins/sentry_mail/models.py

```diff
@@ -266,7 +267,8 @@ def notify(self, notification):
             'X-Sentry-Reply-To': group_id_to_email(group.id),
         }
 
-        for user_id in self.get_send_to(project=project, event=event):
+        send_to_users = users or self.get_send_to(project=project, event=event)
```

Comment:
If an empty list is explicitly passed in as `users` (so, "send to no users") it'll be disregarded and the notification will be sent to any users returned by `self.get_send_to` anyway.

Reply:
Hmm.. I didn't initially realize passing in an empty list was an option. Good catch.